iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/ImportInformationSpaceJobUtils.java

```diff
@@ -0,0 +1,80 @@
+package eu.dnetlib.iis.wf.importer.infospace;
+
+import eu.dnetlib.iis.common.InfoSpaceConstants;
+import eu.dnetlib.iis.common.schemas.IdentifierMapping;
+import eu.dnetlib.iis.common.spark.avro.AvroDataFrameSupport;
+import eu.dnetlib.iis.common.spark.avro.AvroDatasetSupport;
+import org.apache.spark.api.java.JavaRDD;
+import org.apache.spark.sql.*;
+
+import static org.apache.spark.sql.functions.*;
+
+/**
+ * Common utilities used in {@link ImportInformationSpaceJob}.
+ */
+public class ImportInformationSpaceJobUtils {
+
+    private ImportInformationSpaceJobUtils() {
+
+    }
+
+    /**
+     * Produces a mapping from graph id to object store id.
+     * <p>
+     * The object store ids are build for {@link eu.dnetlib.dhp.schema.oaf.Result} records as a mapping from graph ids to
+     * {@link eu.dnetlib.dhp.schema.oaf.Result#originalId} fields. If an entity is deduplicated the graph ids are further
+     * replaced with the deduplicated ids.
+     * <p>
+     * NOTE: this implementation servers as a temporary solution and should be unnecessary when the graph original ids
+     * and object store ids will be the same.
+     *
+     * @param dedupRelation              {@link IdentifierMapping} RDD containing mappings from deduplicated ids to original ids.
+     * @param sourceDataset              {@link eu.dnetlib.dhp.schema.oaf.Dataset} RDD.
+     * @param sourceOtherResearchProduct {@link eu.dnetlib.dhp.schema.oaf.OtherResearchProduct} RDD.
+     * @param sourcePublication          {@link eu.dnetlib.dhp.schema.oaf.Publication} RDD.
+     * @param sourceSoftware             {@link eu.dnetlib.dhp.schema.oaf.Software} RDD.
+     * @param spark                      Instance of SparkSession.
+     * @return An RDD of {@link IdentifierMapping} with object store ids.
+     */
+    public static JavaRDD<IdentifierMapping> produceObjectStoreId(JavaRDD<eu.dnetlib.dhp.schema.oaf.Dataset> sourceDataset,
```

Comment:
I'd rename this method. Mostly because we struggle with a lot of confusion with this whole task and someone reading this code for the first time would be double-confused. Something like `produceGraphIdToObjectStoreIdMapping` or similar would be OK, just to indicate what it really does. I know it is already in method description but it will be easier to understand the code referencing this method.